rackcity/views/it_instance_views.py

```diff
@@ -80,5 +106,25 @@ def instance_add(request):
     failure_message = "Request was invalid. " + failure_message
     return JsonResponse(
         {"failure_message": failure_message},
-        status=HTTPStatus.NOT_ACCEPTABLE,
+        status=HTTPStatus.BAD_REQUEST,
     )
+
+
+@api_view(['GET'])
+@permission_classes([IsAuthenticated])
+def instance_page_count(request):
+    """
+    Return total number of pages according to page size, which must be
+    specified as query parameter.
+    """
+
+    if not request.query_params.get('page_size') or int(request.query_params.get('page_size')) <= 0:
+        return JsonResponse(
+            {"failure_message": "Must specify positive integer page_size."},
+            status=HTTPStatus.BAD_REQUEST,
+        )
+
+    page_size = int(request.query_params.get('page_size'))
+    instance_count = ITInstance.objects.all().count()
+    page_count = math.ceil(instance_count / page_size)
+    return JsonResponse({"pages": page_count})
```

Comment:
This is picky but would it be more clear to say `{"page_count": page_count}` or `{"num_pages": page_count}`? I'll approve anyway

Reply:
Very much agree changing